pytomorrowio/pytomorrowio.py

```diff
@@ -183,177 +177,225 @@ async def _make_call(
             raise InvalidAPIKeyException(resp_json, resp.headers)
         if resp.status == HTTPStatus.TOO_MANY_REQUESTS:
             raise RateLimitedException(resp_json, resp.headers)
-        raise UnknownException(resp_json, resp.headers)
 
-    async def realtime(self, fields: List[str]) -> Dict[str, Any]:
+        try:
+            resp.raise_for_status()
+        except ClientResponseError as error:
+            raise UnknownException(resp_json, resp.headers) from error
+
+        return {}
+
+    async def realtime(
+        self, fields: List[str], reset_num_api_requests: bool = True
+    ) -> Dict[str, Any]:
         """Return realtime weather conditions from Tomorrow.io API."""
-        self._num_api_requests = 0
-        return await self._call_api(
+        if reset_num_api_requests:
+            self._num_api_requests = 0
+
+        data = await self._call_api(
             {
-                "fields": process_v4_fields(fields, REALTIME),
                 "timesteps": ["current"],
+                "fields": fields,
             }
         )
+        if (
+            "data" in data
+            and "timelines" in data["data"]
+            and "intervals" in data["data"]["timelines"][0]
+            and "values" in data["data"]["timelines"][0]["intervals"][0]
+        ):
+            return data["data"]["timelines"][0]["intervals"][0]["values"]
+
+        return {}
 
     async def _forecast(
```

Comment:
What do you think about replacing all of the forecast methods with a single method? The only public methods would be `realtime` and `forecast` methods.

Note: accepting `Sequence` as input values allows for types other than `list` such as `set` and generators.

```
    async def forecast(
        self,
        time_steps: Union[timedelta, Sequence[timedelta]],
        fields: Sequence[str],
        start_time: Optional[datetime] = None,
        duration: Optional[timedelta] = None,
        reset_num_api_requests: Optional[bool] = True,
        **kwargs,
    ) -> Dict[str, Any]:
        """Return nowcast, hourly, and daily forecasts.

        Args:
            time_steps (Union[timedelta, Sequence[timedelta]]): One or more time steps to return.
            fields (Sequence[str]): Fields to return for all time steps.
            start_time (Optional[datetime], optional): Forecast start time. Defaults to now.
            duration (Optional[timedelta], optional): Forecast duration. Defaults to maximum available.
            reset_num_api_requests (Optional[bool], optional): Reset the number of API requests. Defaults to True.

        Returns:
            Dict[str, Any]: Dictionary containing forecasts for the requested time steps.
        """
        if isinstance(time_steps, timedelta):
            time_steps = [time_steps]

        if reset_num_api_requests:
            self._num_api_requests = 0

        params = {
            "timesteps": [self._timestep_to_str(time_step) for time_step in time_steps],
            "fields": fields,
            **kwargs,
        }

        if start_time:
            if not start_time.tzinfo:
                start_time.replace(tzinfo=timezone.utc)
            params["start_time"] = start_time.replace(microsecond=0).isoformat()
        else:
            start_time = datetime.now(tz=timezone.utc)

        if duration:
            end_time = (start_time + duration).replace(microsecond=0)
            params["endTime"] = end_time.isoformat()

        data = await self._call_api(params)

        forecasts: Dict[str, Any] = {}
        try:
            for timeline in data["data"]["timelines"]:
                key = self._timestep_to_result(timeline["timestep"])
                forecasts[key] = timeline["intervals"]
        except LookupError:
            pass  # maybe raise exception instead of silent failure

        return forecasts

    @staticmethod
    def _timestep_to_str(time_step: timedelta) -> str:
        if time_step == ONE_DAY:
            return TIMESTEP_DAILY
        if time_step == ONE_HOUR:
            return TIMESTEP_HOURLY
        if time_step < ONE_HOUR:
            return f"{int(time_step.total_seconds()/60)}m"
        raise ValueError(f"Invalid `time_step` value {time_step}")

    @staticmethod
    def _timestep_to_result(time_step: str) -> str:
        return (
            DAILY
            if time_step == TIMESTEP_DAILY
            else HOURLY
            if time_step == TIMESTEP_HOURLY
            else NOWCAST
        )
```

Reply:
these helper functions are a great idea, thanks! Re: the `forecast` function, I'm OK with making `forecast` a public function instead of a private but there's no harm IMO to keep the other wrapper functions to make things easy. Practically speaking we only use one of the functions here publicly, but it just makes it easier to use the library if someone wants to. Same reason why I have the sync class